src/redux/userLists.js

```diff
@@ -0,0 +1,178 @@
+const USER_LISTS_CLEAR_STATE = 'userLists/USER_LISTS_CLEAR_STATE';
+const USER_LISTS_SET_SELECTED_LIST = 'userLists/USER_LISTS_SET_SELECTED_LIST';
+const FAVORITES_LIST_ID = 'favorites';
+// -- Actions --------------------------------------------------------------- //
+export const userListsLoadState = () => async (dispatch, getState) => {
+  const { network } = getState().settings;
+
+  dispatch({ type: USER_LISTS_LOAD_REQUEST });
+  try {
+    const defaultLists = DefaultTokenLists[network] || [];
+    const userLists = await getUserLists(network);
+    const lists = userLists?.length ? userLists : defaultLists;
+    let allAddresses = [];
+    lists.forEach(list => {
+      allAddresses = [...allAddresses, ...list.tokens];
+    });
+
+    dispatch({
+      payload: { lists },
+      type: USER_LISTS_LOAD_SUCCESS,
+    });
+    const selectedUserList = (await getSelectedUserList()) || FAVORITES_LIST_ID;
+    dispatch(userListsSetSelectedList(selectedUserList, false));
+
+    // Wait until the socket is ready
+    setTimeout(() => {
+      dispatch(emitAssetRequest(allAddresses));
+      dispatch({
+        payload: true,
+        type: USER_LISTS_READY,
+      });
+    }, 3000);
+  } catch (error) {
+    dispatch({ type: USER_LISTS_LOAD_FAILURE });
+  }
+};
+
+export const userListsResetState = () => dispatch =>
+  dispatch({ type: USER_LISTS_CLEAR_STATE });
+
+export const userListsSetSelectedList = (listId, save = true) => dispatch => {
+  dispatch({
+    payload: listId,
+    type: USER_LISTS_SET_SELECTED_LIST,
+  });
+  if (save) {
+    InteractionManager.runAfterInteractions(() => {
+      saveSelectedUserList(listId);
+    });
+  }
+};
+export const userListsClearList = listId => (dispatch, getState) => {
+  const { lists } = getState().userLists;
+  const allNewLists = [...lists];
+
+  // Find the list index
+  let listIndex = null;
+  allNewLists.find((list, index) => {
+    if (list.id === listId) {
+      listIndex = index;
+      return true;
+    }
+    return false;
+  });
+
+  // update the list
+  const newList = { ...allNewLists[listIndex] };
+  newList.tokens = [];
+  allNewLists[listIndex] = newList;
+
+  dispatch({
+    payload: allNewLists,
+    type: USER_LISTS_UPDATE_LISTS,
+  });
+  saveUserLists(allNewLists);
+};
+
+export const userListsUpdateList = (assetAddress, listId, add = true) => (
+  dispatch,
+  getState
+) => {
+  const address = toLower(assetAddress);
+  if (listId === FAVORITES_LIST_ID) {
+    dispatch(uniswapUpdateFavorites(assetAddress, add));
+  } else {
+    const { lists } = getState().userLists;
+    const allNewLists = [...lists];
+
+    // Find the list index
+    let listIndex = null;
+    allNewLists.find((list, index) => {
+      if (list.id === listId) {
+        listIndex = index;
+        return true;
+      }
+      return false;
+    });
+    // Normalize the list
+    const normalizedListTokens = map(allNewLists[listIndex].tokens, toLower);
```

Comment:
Do we need to do this every time if we always add or remove the lowered address?